osu.Game/Rulesets/Mods/IHasFailCondition.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Game.Rulesets.Judgements;
+
+namespace osu.Game.Rulesets.Mods
+{
+    /// <summary>
+    /// Interface for a <see cref="Mod"/> with fail condition
+    /// </summary>
+    public interface IHasFailCondition
+    {
+        bool FailCondition(JudgementResult result);
```

Comment:
Probably carry the documentation in `ModFailCondition` over to here.

Reply:
carried it